geonode/api/resourcebase_api.py

```diff
@@ -287,6 +287,8 @@ def format_objects(self, objects):
             if formatted_obj.get("metadata", None):
                 formatted_obj["metadata"] = [model_to_dict(_m) for _m in formatted_obj["metadata"]]
 
+            formatted_obj["detail_url"] = obj.detail_url
```

Comment:
The detail URL is a property, so is not conerted by the `model_to_dict` function